src/server/string_family.cc

```diff
@@ -1170,6 +1258,75 @@ auto StringFamily::OpMGet(bool fetch_mcflag, bool fetch_mcver, const Transaction
   return response;
 }
 
+/* CL.THROTTLE <key> <max_burst> <count per period> <period> [<quantity>] */
+void StringFamily::ClThrottle(CmdArgList args, ConnectionContext* cntx) {
+  const string_view key = ArgS(args, 1);
+
+  // Allow max burst in number of tokens
+  uint64_t max_burst;
+  const string_view max_burst_str = ArgS(args, 2);
+  if (!absl::SimpleAtoi(max_burst_str, &max_burst)) {
+    return (*cntx)->SendError(kInvalidIntErr);
+  }
+
+  // Emit count of tokens per period
+  uint64_t count;
+  const string_view count_str = ArgS(args, 3);
+  if (!absl::SimpleAtoi(count_str, &count)) {
+    return (*cntx)->SendError(kInvalidIntErr);
+  }
+
+  // Period of emitting count of tokens
+  uint64_t period;
+  const string_view period_str = ArgS(args, 4);
+  if (!absl::SimpleAtoi(period_str, &period)) {
+    return (*cntx)->SendError(kInvalidIntErr);
+  }
+
+  // Apply quantity of tokens now
+  uint64_t quantity = 1;
+  if (args.size() > 5) {
+    const string_view quantity_str = ArgS(args, 5);
+
+    if (!absl::SimpleAtoi(quantity_str, &quantity)) {
+      return (*cntx)->SendError(kInvalidIntErr);
+    }
+  }
+
+  const uint64_t limit = max_burst + 1;
+  const int64_t emission_interval_ms = period * 1000 / count;
```

Comment:
nit: i think it also should be uint64_t and all its derived variables.

Reply:
Changing `emission_interval_ms` to `uint64_t` causes test to fail. Algo is operating on negatives in some cases.

Reply:
hmm, ok.